### PR TITLE
Refactor Honesty

### PR DESCRIPTION
**Pull request by ghost**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Hello,

* Removed unused prop
* Converted the Honesty component into a functional one

Please review :)

### CODE REVIEW

File: client/src/components/settings/Honesty.js [hunk 1]
```diff
@@ -34,23 +33,19 @@ class Honesty extends Component {
...
-}
+  return (
+    <section className='honesty-policy'>
+      <SectionHeader>Academic Honesty Policy</SectionHeader>
+      <FullWidthRow>
+        <Panel className='honesty-panel'>
+          <HonestyPolicy />
+        </Panel>
+        <br />
+        {isHonest ? renderIsHonestAgreed() : renderAgreeButton()}
```
Comment (ojeytonwilliams) [inline client/src/components/settings/Honesty.js:44]:
```suggestion
        {isHonest ? <IsHonestAgreed /> : <AgreeButton />}
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks @max-voronov, this is a nice improvement.  I've suggested a couple of additional changes to make it look a little more idiomatic.

**Comment by max-voronov** [comment]
> Thanks @max-voronov, this is a nice improvement. I've suggested a couple of additional changes to make it look a little more idiomatic.

Thanks for the review 🎉 

Please see 4709f28f8516b7e26f94b17d56e339b0a079bf95 which contains the requested changes.

Please let me know if anything else needed :)

**Review by ojeytonwilliams — DISMISSED** [review]
Thanks for adding those changes.  LGTM :+1:

**Comment by ojeytonwilliams** [comment]
@max-voronov let me re-work the tests - they shouldn't need updating when the change is a pure refactor.  I'll make sure my suggestions didn't break anything and, if so, make the tests less brittle.

**Comment by max-voronov** [comment]
> @max-voronov let me re-work the tests - they shouldn't need updating when the change is a pure refactor. I'll make sure my suggestions didn't break anything and, if so, make the tests less brittle.

I think it fails because before the change both `IsHonestAgreed` and `AgreeButton` were functions and not components, so their contents were part of the snapshot. Now `IsHonestAgreed` and `AgreeButton` are components so their contents should not be part of the snapshot.

**Comment by max-voronov** [comment]
@ojeytonwilliams I've ran `npm test -- -u` and got the following changes: 26451db4d568523eda4d6b79c1535cf7d21417e2

I also think it's maybe better to move both `AgreeButton` and `IsHonestAgreed` outside of `Honesty` because right now we have 2 components that are defined inside a component.

WDYT?

**Review by ojeytonwilliams — DISMISSED** [review]
I think this is fine as is.  Ideally the tests shouldn't care how you structure a component, since it has no impact on what a user interacts with.  That said, we're looking to improve how we test things, https://github.com/freeCodeCamp/freeCodeCamp/issues/39574, so we can come back to this component when we do that.

Regarding `AgreeButton` and `IsHonestAgreed`, we only use them here, so I think it's less cluttered to define them in Honesty.js

**Comment by max-voronov** [comment]
> I think this is fine as is. Ideally the tests shouldn't care how you structure a component, since it has no impact on what a user interacts with. That said, we're looking to improve how we test things, #39574, so we can come back to this component when we do that.
> 
> Regarding `AgreeButton` and `IsHonestAgreed`, we only use them here, so I think it's less cluttered to define them in Honesty.js

After a second thought I don't  think that `AgreeButton` and `IsHonestAgreed` should be components.
Those are just 2 buttons that are being rendered conditionally.

I've updated the PR to render them conditionally without calling functions as it used to be. Now the tests doesn't need the change too.

WDYT?

**Review by ojeytonwilliams — APPROVED** [review]
This LGTM, too. Nice job.

**Review by moT01 — APPROVED** [review]
LGTM @max-voronov 🎉 Thanks for contributing to freeCodeCamp 🎉 We look forward to future contributions 🎉 

For future reference, [this area of the docs](https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally?id=making-changes-locally) recommends a naming convention for branches, and [this area](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) recommends conventions for PR titles and commit messages. No big deal, just an FYI.

Side observation: Should resetting my account unaccept the honesty policy? It doesn't currently.

**Comment by max-voronov** [comment]
> LGTM @max-voronov 🎉 Thanks for contributing to freeCodeCamp 🎉 We look forward to future contributions 🎉
> 
> For future reference, [this area of the docs](https://contribute.freecodecamp.org/#/how-to-setup-freecodecamp-locally?id=making-changes-locally) recommends a naming convention for branches, and [this area](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) recommends conventions for PR titles and commit messages. No big deal, just an FYI.
> 

Thanks you for the review and the links 🎉 

> Side observation: Should resetting my account unaccept the honesty policy? It doesn't currently.

Probably it is. I'll look into it, thanks :)

**Comment by moT01** [comment]
> Side observation: Should resetting my account unaccept the honesty policy? It doesn't currently.

You don't necessarily need to look into that, perhaps it's intended. But it does kind of seem like resetting your account should unaccept it in my opinion. It's pretty minor - you could make an issue if you want @max-voronov so it can be discussed. Not necessary if don't want to.

**Pull request merged**